src/main/java/com/cloudant/search3/AbstractServer.java

```diff
@@ -0,0 +1,46 @@
+package com.cloudant.search3;
+
+import io.netty.bootstrap.ServerBootstrap;
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelHandler;
+import io.netty.channel.ChannelOption;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.nio.NioEventLoopGroup;
+import io.netty.channel.socket.nio.NioServerSocketChannel;
+import io.netty.handler.logging.LogLevel;
+import io.netty.handler.logging.LoggingHandler;
+
+public abstract class AbstractServer {
+
+    private final int port;
+    private final ChannelHandler channelHandler;
+
+    private EventLoopGroup bossGroup;
+    private EventLoopGroup workerGroup;
+    private Channel ch;
+
+    protected AbstractServer(final int port, final ChannelHandler channelHandler) {
+        this.port = port;
+        this.channelHandler = channelHandler;
+    }
+
+    public final void start() throws Exception {
+        bossGroup = new NioEventLoopGroup(1);
+        workerGroup = new NioEventLoopGroup();
+        final ServerBootstrap b = new ServerBootstrap();
+        b.option(ChannelOption.SO_BACKLOG, 1024);
```

Comment:
probably make the number of connections configurable at some point.

Reply:
sure but this is not that setting, this is the maximum length of the queue for accepting new connections. I don't expect this to need altering.